packages/core/src/components/advanced-select/advanced-select.tsx

```diff
@@ -217,13 +219,15 @@ const OptionElement: FunctionalComponent<{
   option: AdvancedSelectOption<never>;
   active: AdvancedSelectOption<never> | undefined;
   activeHint: string | undefined;
+  isPlaceholder: boolean;
   callback: (event: MouseEvent, value: AdvancedSelectOption<never>) => void;
-}> = ({ option, active, activeHint, callback }) => (
+}> = ({ option, active, activeHint, isPlaceholder, callback }) => (
   <li>
     <button
-      class={clsx(["option", { "option-active": active === option }])}
+      class={clsx(["option", { "option-active": active === option, "option-placeholder": isPlaceholder }])}
       type="button"
-      onClick={(e) => callback(e, option)}
+      tabindex={isPlaceholder ? -1 : 0}
+      onClick={(e) => !isPlaceholder && callback(e, option)}
```

Comment:
Dit is een antipatroon: De button bestaat nog steeds, maar doet niets.

De Accessibility inspector toont deze knop ook nog gewoon:

![image](https://github.com/dso-toolkit/dso-toolkit/assets/9696261/26ace963-d336-4dd5-a78f-64624ab0a356)

Ik denk dat `OptionElement` te groot is geworden. Het tonen van de placeholder moet hier buiten blijven. Ik stel voor dat je naast `OptionElement` een nieuw FunctionalComponent maakt, speciaal voor de placeholder.

Idem voor de modellering: Een `AdvancedSelectionOption` kan semantisch immers geen "placeholder" zijn, het is een _optie_.

Verder zie ik nu dat het maken van de `<li>` onderdeel van het FunctionalComponent is geworden. `ul/ol` zijn zo strict met elkaar verbonden, ik zie deze graag naast elkaar staan.

Succes!

Reply:
Goede punten! Ik heb je feedback verwerkt. M'n reacties:

> Dit is een antipatroon: De button bestaat nog steeds, maar doet niets.
> 
> ...
>
> Ik denk dat `OptionElement` te groot is geworden. Het tonen van de placeholder moet hier buiten blijven. Ik stel voor dat je naast `OptionElement` een nieuw FunctionalComponent maakt, speciaal voor de placeholder.

Goed punt, ik heb dit ook even gedaan voor het `redirect` gedeelte, nu hebben we een `FunctionalComponent` voor alle elementen binnen een groep.

> Idem voor de modellering: Een `AdvancedSelectionOption` kan semantisch immers geen "placeholder" zijn, het is een _optie_.

Ik heb hier opnieuw over nagedacht en heb de `placeholder` property verplaatst naar de `AdvancedSelectGroup`. Dit lijkt mij logischer omdat dit gebruikt kan worden in plaats van een lijst van opties. Ik denk dat een toekomstige `disabled` property meer van toepassing gaat zijn om een enkele optie in een lijst van opties uit te schakelen. Dat ligt ook meer in lijn met de specificatie van `<option>`. Ben benieuwd wat je hier van vindt.

> Verder zie ik nu dat het maken van de `<li>` onderdeel van het FunctionalComponent is geworden. `ul/ol` zijn zo strict met elkaar verbonden, ik zie deze graag naast elkaar staan.

Kheb de `<li>` uit de `OptionElement` verplaatst, dit ziet er inderdaad wat logischer uit.

Reply:
> > Dit is een antipatroon: De button bestaat nog steeds, maar doet niets.
> > ...
> > Ik denk dat `OptionElement` te groot is geworden. Het tonen van de placeholder moet hier buiten blijven. Ik stel voor dat je naast `OptionElement` een nieuw FunctionalComponent maakt, speciaal voor de placeholder.
> 
> Goed punt, ik heb dit ook even gedaan voor het `redirect` gedeelte, nu hebben we een `FunctionalComponent` voor alle elementen binnen een groep.

Ziet er heel charmant uit 👍 

> > Idem voor de modellering: Een `AdvancedSelectionOption` kan semantisch immers geen "placeholder" zijn, het is een _optie_.
> 
> Ik heb hier opnieuw over nagedacht en heb de `placeholder` property verplaatst naar de `AdvancedSelectGroup`. Dit lijkt mij logischer omdat dit gebruikt kan worden in plaats van een lijst van opties.

![image](https://github.com/dso-toolkit/dso-toolkit/assets/9696261/d13d317a-c1a1-44a1-b085-1f46ce5cc861)

Maar ik zie toch wel wat pijnlijke constructies in de code terugkomen, met name bij het testen in welke modus het item zich bevindt. En die return type van de `map()` naar badges voor summary counters zit mij ook niet helemaal lekker 🤔 

```ts
export interface AdvancedSelectGroup<T> {
  label: string;
  summaryCounter?: boolean;
  redirect?: AdvancedSelectGroupRedirect;
  options: AdvancedSelectOption<T>[]; // weer verplichte property
  variant?: AdvancedSelectVariant;
}

export interface AdvancedSelectPlaceholder {
  label: string;
  summaryCounter?: boolean; // Of misschien niet meer nodig?
  redirect?: AdvancedSelectGroupRedirect;
  placeholder: string;
  variant?: AdvancedSelectVariant;
}

export type AdvancedSelectOptionOrGroup<T> = AdvancedSelectOption<T> | AdvancedSelectGroup<T>; | AdvancedSelectPlaceholder;
```

* Expliciete modellering zorgt ervoor dat je placeholder en options niet kan combineren...
* ...waardoor de code zal simpeler wordt: De enige aanname is `"

Enfin, wat denk je?

> Ik denk dat een toekomstige `disabled` property meer van toepassing gaat zijn om een enkele optie in een lijst van opties uit te schakelen. Dat ligt ook meer in lijn met de specificatie van `<option>`. Ben benieuwd wat je hier van vindt.

Ik ben even kwijt waar `disabled` vandaan komt, maar ik ben het helemaal met je eens.

> > Verder zie ik nu dat het maken van de `<li>` onderdeel van het FunctionalComponent is geworden. `ul/ol` zijn zo strict met elkaar verbonden, ik zie deze graag naast elkaar staan.
> 
> Kheb de `<li>` uit de `OptionElement` verplaatst, dit ziet er inderdaad wat logischer uit.

Mooi man!